### PR TITLE
Some fixes to federation.

### PR DESCRIPTION
**Pull request by dessalines**
- Advanced code migrations now disable then re-enable triggers.
  Brings run time down to < 15 seconds, no need to thread them.
- Changing ap_ids and actor_ids in migrations to a fake url,
  so it doesn't break XsdAnyUri in activitystreams.

### CODE REVIEW

**Comment by dessalines** [comment]
Just needs a quick look over, I tested with production data, and its able to do the code migrations / key generations in less than 15 seconds.

File: docker/federation/docker-compose.yml [hunk 1]
```diff
@@ -18,7 +18,6 @@ services:
       - lemmy_gamma
       - pictrs_gamma
       - iframely
-    restart: "always"
```
Comment (dessalines) [inline docker/federation/docker-compose.yml:21]: Having all these instances run on startup locally was really annoying.

File: server/src/db/code_migrations.rs [hunk 2]
```diff
@@ -32,10 +31,12 @@ fn user_updates_2020_04_02(conn: &PgConnection) -> Result<(), Error> {
 
   // Update the actor_id, private_key, and public_key, last_refreshed_at
   let incorrect_users = user_
-    .filter(actor_id.eq("changeme"))
+    .filter(actor_id.eq("http://fake.com"))
     .filter(local.eq(true))
     .load::<User_>(conn)?;
 
+  sql_query("alter table user_ disable trigger refresh_user").execute(conn)?;
+
```
Comment (dessalines) [inline server/src/db/code_migrations.rs:39]: The main stuff here, disabling the re-enabling the triggers.

File: docker/federation/docker-compose.yml (hunk 1 above)
Comment (Nutomic) [reply docker/federation/docker-compose.yml:21]: You mean it takes too long to start Lemmy? For that it should help to remove iframely from depends_on, anyway its not needed for Lemmy to start.

Comment (dessalines) [reply docker/federation/docker-compose.yml:21]: Nah just that on restarting my docker, it automatically restarts all these, and that was kinda annoying.

**Comment by dessalines** [comment]
> Setting the apub_id like this is quite hacky though, would be much cleaner to make it nullable instead.

It really needs to be `not null unique` (every user does need an actor_id), but since the actor_id column has to be generated in code, I can't do that from a sql migration initially (then I can add a uniqueness constraint at some point).

**Comment by Nutomic** [comment]
Okay if we can completely remove `http://fake.com` from the code soon, I am fine with this.

**Pull request merged**